packages/yourloops/lib/auth/hook.tsx

```diff
@@ -280,14 +280,20 @@ function AuthContextImpl(api: AuthAPI): AuthContext {
     return api.resetPassword(key, username, password, traceToken);
   };
 
-  const switchRoleToHCP = async (): Promise<void> => {
+  const switchRoleToHCP = async (accept: boolean): Promise<void> => {
     const authInfo = await getAuthInfos();
     if (authInfo.user.role !== UserRoles.caregiver) {
       throw new Error("invalid-user-role");
     }
 
-    await api.updateUser(authInfo, { role: UserRoles.hcp });
+    if (authInfo.user?.profile !== undefined) {
+      const now = new Date().toISOString();
+      authInfo.user.profile.termsOfUse = { AcceptanceDate: now, IsAccepted: accept };
+      authInfo.user.profile.privacyPolicy = { AcceptanceDate: now, IsAccepted: accept };
+      await api.updateProfile(authInfo);
+    }

```

Comment:
In conjunction with https://github.com/mdblp/blip/pull/499:
```suggestion
    const now = new Date().toISOString();
    const profile = authInfo.user.profile ?? {} as Profile;
    profile.termsOfUse = { AcceptanceDate: now, IsAccepted: true };
    profile.privacyPolicy = { AcceptanceDate: now, IsAccepted: true };
    const updatedProfile = await updateProfile(profile, false);
```

And below:
```
const updatedUser: User = { ...authInfo.user, role: UserRoles.hcp, profile: updatedProfile };
```

Reply:
as dicussed will be change in #499